### ISSUE TITLE
Beam search in dgl transformer

### ISSUE THREAD

**Issue by YongtaoGE** [mandubian/pytorch_math_dataset#1]
Really nice work! Have you found the reason why beam search fail in dgl transformer?

**Comment by mandubian** [comment]
Thks ;)
No, I haven't yet searched for that, I've been working on other topics lately.
But I'm not sure if it's an issue with beam search or just the model being limited in its learnt notions.
(Let say it's not so fast to train with my own GPU)

**Comment by YongtaoGE** [comment]
I use this code to train the whole dataset with basic transformer model. The training process seems correct as the loss keeps decreasing. But I found that the model can't get the right answer of basic arithmetic add_sub_multiple questions like "Evaluate 1 + 1 + (4 - 7) - -2."  In origin paper, this sub module get very high accuracy. So have you also met that problem.

 
<img width="865" alt="loss log" src="https://user-images.githubusercontent.com/22744013/61876997-b3b04a00-af20-11e9-92b5-874164fcce08.png">

**Comment by mandubian** [comment]
Do you mean that using a normal transformer, you have issues with beam search and not only with DGL transformer? If yes, it would help searching the cause because I had no idea yet...

**Comment by YongtaoGE** [comment]
yep! some modules work fine while others not. I could not locate the bug.
